### PR TITLE
[SPARK-10280][MLlib][PySpark][Docs] Add @since annotation to pyspark.ml.classification

### PR DESCRIPTION
**Pull request by yu-iskw**

### CODE REVIEW

**Comment by noel-smith** [comment]
Should replace `.. addedversion::` with `.. versionadded::` in all the class docstrings.

**Comment by noel-smith** [comment]
Some functions have been removed since this PR was created + there are some new classes (`MultilayerPerceptronClassifier`) without version numbers.

@mengxr How do you usually handle this? Would you expect the PR to be rebased/merged with the latest version of the file?

**Comment by yu-iskw** [comment]
@noel-smith @mengxr I have merged this PR with master and have added the version info. to MultilayerPerceptronClassifier and MultilayerPerceptronClassificationModel which were added after sending this PR. Could you review it again please?

**Comment by mengxr** [comment]
@noel-smith New PRs should come along with `@since`. I forgot that for MLPC Python API. @yu-iskw Could you rebase master?

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -116,6 +120,37 @@ def setParams(self, featuresCol="features", labelCol="label", predictionCol="pre
...
+        self._paramMap[self.fitIntercept] = value
+        return self
+
+    @since("1.4.0")
+    def getFitIntercept(self):
+        """
+        Gets the value of fitIntercept or its default value.
+        """
+        return self.getOrDefault(self.fitIntercept)
+
```
Comment (noel-smith) [inline python/pyspark/ml/classification.py]: Should remove the four functions above - they were deleted by a recent commit to master.

**Comment by noel-smith** [comment]
The comments on `MultilayerPerceptronClassifier` + `MultilayerPerceptronClassifierModel` are good. Just need to remove the four functions noted above (these were deleted recently in master).

**Comment by yu-iskw** [comment]
Jenkins, test this please.

**Comment by yu-iskw** [comment]
Jenkins, test this please.

**Comment by yu-iskw** [comment]
@noel-smith I have rebase and remove the methods which had been already removed. Could you review it?

**Comment by noel-smith** [comment]
@yu-iskw Thanks for removing those methods - but did you loose some of the version numbers when rebasing? I.e. ```DecisionTreeClassifier.getImpurity()``` has a ```@since```, but ```RandomForestClassifier.getImpurity()``` doesn't now.

**Comment by yu-iskw** [comment]
@noel-smith thank you for the review. I added them again. Could you review it?

**Comment by noel-smith** [comment]
@yu-iskw Almost there - you're just missing the version numbers on `GBTClassifier` class + methods (they should all be set to `1.4.0`), everything else looks good though.

**Comment by yu-iskw** [comment]
@noel-smith thank you for the review. I rebase this PR with the mater and then updated two parts.

- Add `@since` tags to `BGTClassifier` class and object
- Add `@since` tags to `TreeClassifierParams`

**Comment by yu-iskw** [comment]
Jenkins, test this please.

**Comment by noel-smith** [comment]
@yu-iskw Thanks for making the changes - LGTM

**Comment by yu-iskw** [comment]
@noel-smith thanks you for the review.

**Comment by yu-iskw** [comment]
Jenkins, test this please.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**